Feat/auth

leeyeram84 opened the pull request:
## 🔥 작업 내용

- 로그인 및 회원가입 페이지 기본 input 폼 작성
- 로그인 및 회원가입 페이지 유효성 검사 적용

## 💁🏻‍♀️ 테스트 체크리스트

# 로그인 페이지 
- 로그인 input 창 값 미입력시 하단에 메세지 출력
- 로그인 버튼 입력 시 데이터 받아오기
 <img src ="https://media.discordapp.net/attachments/1244516648866680885/1299186157329973249/2024-10-2510.39.34-ezgif.com-video-to-gif-converter.gif?ex=671c48e5&is=671af765&hm=0e7f13cd1be9e846242ff4f5c78121b61c63005655df9304a2e7796e4927a3d9&=&width=1200&height=750" />

# 회원가입 페이지
- 회원가입 input 별 스케마 적용
- 회원가입 버튼 입력 시 데이터 받아오기
<img src="https://media.discordapp.net/attachments/1244516648866680885/1299188500066861077/2024-10-2510.46.57-ezgif.com-video-to-gif-converter.gif?ex=671c4b14&is=671af994&hm=65ba17107961fcabe2f45841abbdfe4649a06a6afa028159b0403601d43237e8&=&width=1200&height=750" />

## 👍 참고 사항

- 회원가입 시 비밀번호 확인 메세지 관련 이슈
: 비밀번호 확인란 입력 후 기존 비밀번호 인풋 메세지가 실시간으로 변경되지 않음. 금일 중 해결 예정

leeyeram84 committed:
test:test

leeyeram84 committed:
feat : 로그인 베이스 제작

leeyeram84 committed:
feat : 제출버튼 광클 방지 버튼 비활성화

leeyeram84 committed:
feat : 인풋 조건부 추가

zod와 shadcn의 Form을 이용하여 input시 조건부 생성

leeyeram84 committed:
Merge branch 'dev' of https://github.com/seokwon27/takkom into feat/signin

leeyeram84 committed:
add : 로그인/회원가입 기본 폼 작성 및 스케마 작성

leeyeram84 committed:
Merge branch 'dev' of https://github.com/seokwon27/takkom into feat/auth

leeyeram84 committed:
fix : 불필요한 Link 제거

stopbrother reviewed (approved):
page.tsx의 Link들 layout.tsx로 옮기면 좋을 것 같습니다 !

seokwon27 commented on src/components/auth/auth.tsx:
아 파일은 사용하실 예정이실까요?

seokwon27 replied:
버셀 빌드 에러도 해결해야할거같습니다!

leeyeram84 replied:
이후 컴포넌트를 signin / signup 으로 분리 할 때 사용 할 예정입니다!
버셀 빌드도 해결했습니다!

leeyeram84 committed:
fix : shadcn package 수정 및 비밀번호 유효성 검사 메세지 수정

leeyeram84 committed:
fix : 회원가입 비밀번호 스케마 수정 및 return FormField 사용하여 수정

- 회원가입 password 스키마를 삭제하여 UX 상향
- return 문 안에서 Form 사용법이 잘못되어 FormField를 사용하여 수정

joycie416 commented on src/app/(auth)/signup/page.tsx:
```diff
@@ -1,5 +1,11 @@
+import AuthForm from "@/components/auth/AuthForm";
+
 const page = () => {
```
`SignUpPage`로 창식님이 알려주신대로 변경하면 좋을 것 같습니다~

leeyeram84 replied:
변경했습니다! 깜박할뻔했네요ㅠ 감사합니다!

joycie416 commented on src/app/(auth)/signin/page.tsx:
```diff
@@ -1,5 +1,9 @@
+import AuthForm from "@/components/auth/AuthForm";
 
-export default page;
+export default function page() {
```
`SignInPage`로 창식님이 알려주신대로 변경하면 좋을 것 같습니다~

그리고 이부분만 function으로 되어있네요? 화살표 함수가 아니라?

seokwon27 replied:
오 미처 보지 못했네요 GOOD~

joycie416 reviewed (changes requested):
코드 코멘트 확인 부탁드립니다~

leeyeram84 committed:
fix : page 이름 변경

leeyeram84 committed:
fix : 함수 화살표 함수로 변경

leeyeram84 committed:
fix : 컴포넌트 파스칼케이스로 수정
